core-ui/src/state/utils/getFetchFn.ts

```diff
@@ -0,0 +1,25 @@
+import { authDataState } from '../authDataAtom';
+import { clusterState } from '../clusterAtom';
+import { clusterConfigState } from '../clusterConfigAtom';
+import { ssoDataState } from '../ssoDataAtom';
+import { configState } from '../configAtom';
+import { GetRecoilValue } from 'recoil';
+import { createFetchFn } from '../../shared/hooks/BackendAPI/useFetch';
+
+export const getFetchFn = (get: GetRecoilValue) => {
+  const authData = get(authDataState);
+  const cluster = get(clusterState);
+  const clusterConfig = get(clusterConfigState);
+  const ssoData = get(ssoDataState);
+  const { fromConfig } = get(configState) || {};
+
+  if ((authData || ssoData) && cluster && clusterConfig && fromConfig) {
```

Comment:
Only `authData` is required, `ssoData` is optional. Optionally we could check for `ssoData` **only** if SSO_LOGIN feature is enabled?

Reply:
Let's not go for features check atm, as we have the rerender problem there due to Prometheus